src/app/azureResourceIdentifier/services/azureResourceIdentifierService.ts

```diff
@@ -0,0 +1,110 @@
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License
+ **********************************************************/
+import { ContinuingResultSet } from '../../api/models/continuingResultSet';
+import { AzureResourceIdentifier } from '../models/azureResourceIdentifier';
+import { AzureResourceIdentifierType } from '../models/azureResourceIdentifierType';
+import { AzureResourceIdentifierQuery } from '../models/azureResourceIdentifierQuery';
+import { AzureResourceIdentifierQueryResult } from '../models/azureResourceIdentifierQueryResult';
+import { APPLICATION_JSON, HTTP_OPERATION_TYPES } from '../../api/constants';
+import { AzureResourceManagementEndpoint } from '../models/azureResourceManagementEndpoint';
+import { HttpError } from '../../api/models/httpError';
+import { mapPropertyArrayToObject } from '../../api/shared/mapUtils';
+
+const azureResourceManagementAPIVersion = '2019-04-01';
+const azureResourceManagementQueryFields = [
+    'id',
+    'name',
+    'type',
+    'location',
+    'resourceGroup',
+    'subscriptionId'
+];
+
+export interface GetAzureResourceIdentifiersParameters {
+    azureResourceManagementEndpoint: AzureResourceManagementEndpoint;
+    continuationToken?: string;
+    subscriptionIds: string[];
+    resourceType: AzureResourceIdentifierType;
+}
+export const getAzureResourceIdentifiers = async (parameters: GetAzureResourceIdentifiersParameters): Promise<ContinuingResultSet<AzureResourceIdentifier>> => {
+    const { azureResourceManagementEndpoint, continuationToken, subscriptionIds, resourceType } = parameters;
+    const { authorizationToken, endpoint } = azureResourceManagementEndpoint;
+
+    const resourceUrl = `https://${endpoint}/providers/Microsoft.ResourcesGraph/resources?api-version=${azureResourceManagementAPIVersion}`;
+    const requestBody: AzureResourceIdentifierQuery = {
+        query: `where type =~ '${resourceType}' | project ${azureResourceManagementQueryFields.join(',')}`,
+        subscriptions: subscriptionIds,
+    };
+
+    if (continuationToken) {
+        requestBody.options = {
+            $skipToken: continuationToken
+        };
+    }
+
+    const serviceRequestParams: RequestInit = {
+        body: JSON.stringify(requestBody),
+        headers: new Headers({
+            'Accept': APPLICATION_JSON,
+            'Authorization': `Bearer ${authorizationToken}`,
+            'Content-Type': APPLICATION_JSON
+        }),
+        method: HTTP_OPERATION_TYPES.Post
+    };
+    const response = await fetch(resourceUrl, serviceRequestParams);
+    if (!response.ok) {
+        throw new HttpError(response.status);
+    }
+
+    const responseBody = await response.json() as AzureResourceIdentifierQueryResult;
+    const azureResourceIdentifiers = (responseBody.data && responseBody.data.rows) ?
+        responseBody.data.rows.map(row => mapPropertyArrayToObject(azureResourceManagementQueryFields, row) as AzureResourceIdentifier) :
```

Comment:
a fun journey to undersand this logic personally. Is it that we query all the fields, and the returned array 'row' here would have everything in order, and we then map it back to an oject with keys being the qeury fields? Wondering if the helper which checks the length boundary is just an extra check or we actually run into the situation that the two lengths would be different.

Reply:
the helper takes the desiredFields as the canonical value -- if the number of entries in the row exceeds the number of desired fields -- the value is not written to the returned data object.  Conversely, the desiredField is returned with 'undefined' if the value array is out of bounds.